### ISSUE TITLE
TENET.ExperimentHub

### ISSUE THREAD

**Issue by EthanNelson-Moore** [Bioconductor/Contributions#3590]
Update the following URL to point to the GitHub repository of
the package you wish to submit to _Bioconductor_

- Repository: https://github.com/rhielab/TENET.ExperimentHub

Confirm the following by editing each check box to '[x]'

- [x] I understand that by submitting my package to _Bioconductor_,
  the package source and all review commentary are visible to the
  general public.

- [x] I have read the _Bioconductor_ [Package Submission][2]
  instructions. My package is consistent with the _Bioconductor_
  [Package Guidelines][1].

- [x] I understand Bioconductor [Package Naming Policy][9] and acknowledge
  Bioconductor may retain use of package name.

- [x] I understand that a minimum requirement for package acceptance 
  is to pass R CMD check and R CMD BiocCheck with no ERROR or WARNINGS. 
  Passing these checks does not result in automatic acceptance. The 
  package will then undergo a formal review and recommendations for 
  acceptance regarding other Bioconductor standards will be addressed.

- [x] My package addresses statistical or bioinformatic issues related
  to the analysis and comprehension of high throughput genomic data.

- [x] I am committed to the long-term maintenance of my package. This
  includes monitoring the [support site][3] for issues that users may
  have, subscribing to the [bioc-devel][4] mailing list to stay aware
  of developments in the _Bioconductor_ community, responding promptly
  to requests for updates from the Core team in response to changes in
  _R_ or underlying software.
  
- [x] I am familiar with the [Bioconductor code of conduct][7] and 
  agree to abide by it.

I am familiar with the essential aspects of _Bioconductor_ software
management, including:

- [x] The 'devel' branch for new packages and features.
- [x] The stable 'release' branch, made available every six
      months, for bug fixes.
- [x] _Bioconductor_ version control using [Git][5]
  (optionally [via GitHub][6]).

For questions/help about the submission process, including questions about
the output of the automatic reports generated by the SPB (Single Package
Builder), please use the #package-submission channel of our Community Slack.
Follow the link on the home page of the [Bioconductor website][8] to sign up.

[1]: https://contributions.bioconductor.org/
[2]: https://bioconductor.org/developers/package-submission/
[3]: https://support.bioconductor.org
[4]: https://stat.ethz.ch/mailman/listinfo/bioc-devel
[5]: http://bioconductor.org/developers/how-to/git/
[6]: http://bioconductor.org/developers/how-to/git/sync-existing-repositories/
[7]: https://bioconductor.org/about/code-of-conduct/
[8]: https://bioconductor.org/
[9]: https://bioconductor.org/developers/package-submission/#naming

**Comment by EthanNelson-Moore** [comment]
AdditionalPackage: https://github.com/rhielab/TENET

**Comment by EthanNelson-Moore** [comment]
AdditionalPackage: https://github.com/rhielab/TENET

**Comment by EthanNelson-Moore** [comment]
@Kayla-Morrell Hello, how is the review process going? We haven't heard anything back.

**Comment by Kayla-Morrell** [comment]
@EthanNelson-Moore - My apologizes for the late response. I will work on getting the initial review of the packages posted within the next couple of days. Thank you for your patience.

**Comment by EthanNelson-Moore** [comment]
No problem!

**Comment by EthanNelson-Moore** [comment]
Hello, do you have any further updates on the status of the review process?

**Comment by Kayla-Morrell** [comment]
@EthanNelson-Moore Thank you for submitting to _Bioconductor_. Please see the initial review of 
the package below. The required changes must be made while the suggested 
changes do not have to be (though we strongly encourage them). Comment back 
here with updates that have been made and when the package is ready for a 
re-review.

# TENET.ExperimentHub

## The DESCRIPTION file

- [ ] REQUIRED: Update R version dependency from 4.4 to 4.5.0
- [ ] SUGGESTION: Consider adding these automatically suggested biocViews:
	Somatic, CopyNumberVariationData

## R code

- [ ] SUGGESTION: Consider shorter lines; 47 lines are > 80 characters long.

# TENET

## The DESCRIPTION file

- [ ] REQUIRED: Update R version dependency from 4.4 to 4.5.0

## Documentation

- [ ] REQUIRED: All code blocks should be executable (not include eval = FALSE).
- [ ] REQUIRED: The 'data-raw' directory should be moved to an 'inst/script' directory since it's code that creates the data file.

## Unit tests

- [ ] SUGGESTION: Consider adding unit tests. We strongly encourage them. See https://contributions.bioconductor.org/tests.html

## R code

- [ ] SUGGESTION: Avoid 'suppressWarnings'/'*Messages' if possible
- [ ] SUGGESTION: For formatting reasons, consider shorter lines. There are 55 lines that are > 80 characters long.

Best,
Kayla

**Comment by EthanNelson-Moore** [comment]
Hi, Kayla,

It's great to hear from you. I have addressed the issues you mentioned which are realistically solvable; I will explain why the other ones cannot be fixed easily.

> [@EthanNelson-Moore](https://github.com/EthanNelson-Moore) Thank you for submitting to _Bioconductor_. Please see the initial review of the package below. The required changes must be made while the suggested changes do not have to be (though we strongly encourage them). Comment back here with updates that have been made and when the package is ready for a re-review.
> 
> # TENET.ExperimentHub
> ## The DESCRIPTION file
> * [ ]  REQUIRED: Update R version dependency from 4.4 to 4.5.0

I have pushed a commit to the repository addressing this.

> * [ ]  SUGGESTION: Consider adding these automatically suggested biocViews:
>   Somatic, CopyNumberVariationData

We decided not to add these biocViews since, although the package includes example somatic mutation and CNV datasets, they are synthesized and do not represent any real datasets.

> 
> ## R code
> * [ ]  SUGGESTION: Consider shorter lines; 47 lines are > 80 characters long.

I have pushed a commit to the repository addressing most of these issues.
The remaining lines are impossible to shorten because they contain URLs or code syntax requirements prevent them from being shortened.

> 
> # TENET
> ## The DESCRIPTION file
> * [ ]  REQUIRED: Update R version dependency from 4.4 to 4.5.0

I have pushed a commit to the repository addressing this.

> 
> ## Documentation
> * [ ]  REQUIRED: All code blocks should be executable (not include eval = FALSE).

The code blocks with `eval = FALSE` either contain installation instructions, take a very long time to run and would cause the vignette build to exceed the 10 minute time limit, or download data from the Internet, which could also cause the vignette build to time out.

> * [ ]  REQUIRED: The 'data-raw' directory should be moved to an 'inst/script' directory since it's code that creates the data file.

I have pushed a commit to the repository addressing this.

> 
> ## Unit tests
> * [ ]  SUGGESTION: Consider adding unit tests. We strongly encourage them. See https://contributions.bioconductor.org/tests.html

While it would be ideal if TENET had unit tests, this would require a lot of effort due to the large number of functions included in TENET and the large number of options they have. The vignette build process fairly thoroughly tests the functionality of TENET.

> 
> ## R code
> * [ ]  SUGGESTION: Avoid 'suppressWarnings'/'*Messages' if possible

All instances of suppressWarnings have comments explaining why they are necessary. suppressMessages is used to suppress unimportant messages when loading datasets via AnnotationHub and sesameData.

> * [ ]  SUGGESTION: For formatting reasons, consider shorter lines. There are 55 lines that are > 80 characters long.

I have pushed a commit to the repository addressing some of these issues.
The remaining lines are impossible to shorten because they contain URLs or are in generated files.

>
> Best, Kayla

**Comment by EthanNelson-Moore** [comment]
@Kayla-Morrell Just checking in since we haven't heard anything further. What is the current status of the package reviews?

**Comment by EthanNelson-Moore** [comment]
@Kayla-Morrell We are under a journal-imposed deadline for the resubmission of a paper on our new TENET packages, and it cannot be resubmitted until the packages are accepted to Bioconductor. Would you please expedite the review of these packages?

**Comment by lshep** [comment]
I'll look at this today for you.

**Comment by lshep** [comment]
So just to confirm easyTENET runs steps 2-6  as one step?  If so then we can make an exception for the eval=FALSE provide the steps are run on their own to ensure testing.   That being said,  I think the only step that doesn't appear to be evaluated anywhere then is step3GetAnalysisZScores.  Is there a possibility of having that evaluated either in the man page or the vignette to ensure that it is running properly?

**Comment by EthanNelson-Moore** [comment]
@lshep Yes, `easyTENET` runs steps 1-6 as one step. I've just pushed a commit to enable evaluation of step 3 during the vignette build.

**Comment by EthanNelson-Moore** [comment]
@lshep Do you know what could be causing this "abnormal" result?

**Comment by EthanNelson-Moore** [comment]
There are quite a few strange things about the latest [build report](https://bioconductor.org/spb_reports/TENET_buildreport_20250227205324.html):

* The binary build log, BuildBinCommand, and BuildBinTime are empty, and the binary build is marked as "skipped", even though the link above to download the binary package works.
* The package version is 0.0.0.
* The R version, OS, and architecture are blank, and the OS label before the binary download link is missing.

**Comment by EthanNelson-Moore** [comment]
@lshep Is this due to a bug on Bioconductor's build server?

**Comment by lshep** [comment]
yes it looks like our builder crashed last night. I'll reboot everything and kick off a manual build. It should be available shortly

**Comment by EthanNelson-Moore** [comment]
@lshep There still seems to be an issue with the builder.
The binary build information is still missing on https://bioconductor.org/spb_reports/TENET_buildreport_20250228074554.html and it does not seem to be responding to pushes - see https://github.com/Bioconductor/Contributions/issues/3696#issuecomment-2691496925
Could this have something to do with our recent change to use 2 CPU cores in step 3 so the vignette build doesn't time out?

**Comment by lshep** [comment]
BreastSubtypeR is not working because they did not push with a valid version bump.  The system is on and registering build reports.  I will look into the build bin step but I think this is okay and the OS didn't register yet on that report as I kicked it off too soon to reset that section; but the build and check report are accurate and the build product above is downloadable and available.

**Comment by EthanNelson-Moore** [comment]
@lshep Have you had a chance to further review the package code now that the check issues are resolved?

**Comment by lshep** [comment]
The default branch of your GitHub repository has been added to Bioconductor's
git repository as branch devel.

To use the git.bioconductor.org repository, we need an 'ssh' key to associate with your github user name. If your GitHub account already has ssh public keys (https://github.com/EthanNelson-Moore.keys is not empty), then no further steps are required. Otherwise, do the following:

1. [Add an SSH key to your github account][]
1. [Submit your SSH key to Bioconductor][]

See further instructions at

https://bioconductor.org/developers/how-to/git/

for working with this repository. See especially

https://bioconductor.org/developers/how-to/git/new-package-workflow/
https://bioconductor.org/developers/how-to/git/sync-existing-repositories/

to keep your GitHub and Bioconductor repositories in sync.

Your package will be included in the next nigthly 'devel' build (check-out from git at about 6 pm Eastern; build completion around 2pm Eastern the next day) at

https://bioconductor.org/checkResults/

(Builds sometimes fail, so ensure that the date stamps on the main landing page are consistent with the addition of your package). Once the package builds successfully, you package will be available for download in the 'Devel' version of Bioconductor using `BiocManager::install("TENET.ExperimentHub")`. The package 'landing page' will be created at

https://bioconductor.org/packages/TENET.ExperimentHub

If you have any questions, please contact the bioc-devel mailing list (https://stat.ethz.ch/mailman/listinfo/bioc-devel); this issue will not be monitored further.

[Add an SSH key to your github account]: https://help.github.com/articles/connecting-to-github-with-ssh/
[Submit your SSH key to Bioconductor]: https://git.bioconductor.org/BiocCredentials/

**Comment by lshep** [comment]
The default branch of your GitHub repository has been added to Bioconductor's
git repository as branch devel.

To use the git.bioconductor.org repository, we need an 'ssh' key to associate with your github user name. If your GitHub account already has ssh public keys (https://github.com/EthanNelson-Moore.keys is not empty), then no further steps are required. Otherwise, do the following:

1. [Add an SSH key to your github account][]
1. [Submit your SSH key to Bioconductor][]

See further instructions at

https://bioconductor.org/developers/how-to/git/

for working with this repository. See especially

https://bioconductor.org/developers/how-to/git/new-package-workflow/
https://bioconductor.org/developers/how-to/git/sync-existing-repositories/

to keep your GitHub and Bioconductor repositories in sync.

Your package will be included in the next nigthly 'devel' build (check-out from git at about 6 pm Eastern; build completion around 2pm Eastern the next day) at

https://bioconductor.org/checkResults/

(Builds sometimes fail, so ensure that the date stamps on the main landing page are consistent with the addition of your package). Once the package builds successfully, you package will be available for download in the 'Devel' version of Bioconductor using `BiocManager::install("TENET")`. The package 'landing page' will be created at

https://bioconductor.org/packages/TENET

If you have any questions, please contact the bioc-devel mailing list (https://stat.ethz.ch/mailman/listinfo/bioc-devel); this issue will not be monitored further.

[Add an SSH key to your github account]: https://help.github.com/articles/connecting-to-github-with-ssh/
[Submit your SSH key to Bioconductor]: https://git.bioconductor.org/BiocCredentials/
